lib/Sema/ConstraintSystem.h

```diff
@@ -1979,7 +1979,13 @@ class ConstraintSystem {
   /// Lookup and return parent associated with given expression.
   Expr *getParentExpr(Expr *expr) const {
     auto e = ExprWeights.find(expr);
-    return e != ExprWeights.end() ? e->second.second : nullptr;
+    if (e != ExprWeights.end())
+      return e->second.second;
+
+    if (baseCS && baseCS != this)
```

Comment:
Unfortunately it would have to be a `while` because CSDiag could re-typecheck to the arbitrary depth...

Reply:
It's recursive

Reply:
Ah right, sorry it calls `getParentExpr` :)

Reply:
Sorry for the noise, it's been a long week...